interpreter/src/functions.rs

```diff
@@ -310,6 +310,100 @@ pub fn all(
     };
 }
 
+/// Returns a boolean value indicating whether a or more values in the provided
+/// list or map meet the predicate defined by the provided expression. If
+/// called on a map, the predicate is applied to the map keys.
+///
+/// This function is intended to be used like the CEL-go `exists` macro:
+/// https://github.com/google/cel-spec/blob/master/doc/langdef.md#macros
+///
+/// # Example
+/// ```cel
+/// [1, 2, 3].exists(x, x > 0) == true
+/// [{1:true, 2:true, 3:false}].exists(x, x > 0) == true
+/// ```
+pub fn exists(
+    ftx: &FunctionContext,
+    This(this): This<Value>,
+    ident: Identifier,
+    expr: Expression,
+) -> Result<bool> {
+    match this {
+        Value::List(items) => {
+            let mut ptx = ftx.ptx.clone();
+            for item in items.iter() {
+                ptx.add_variable(&ident, item);
+                if let Value::Bool(true) = ptx.resolve(&expr)? {
+                    return Ok(true);
+                }
+            }
+            Ok(false)
+        }
+        Value::Map(value) => {
+            let mut ptx = ftx.ptx.clone();
+            for key in value.map.keys() {
+                ptx.add_variable(&ident, key);
+                if let Value::Bool(true) = ptx.resolve(&expr)? {
+                    return Ok(true);
+                }
+            }
+            Ok(false)
+        }
+        _ => Err(this.error_expected_type(ValueType::List)),
+    }
+}
+
+/// Returns a boolean value indicating whether only one value in the provided
+/// list or map meets the predicate defined by the provided expression. If
+/// called on a map, the predicate is applied to the map keys.
+///
+/// This function is intended to be used like the CEL-go `exists` macro:
+/// https://github.com/google/cel-spec/blob/master/doc/langdef.md#macros
+///
+/// # Example
+/// ```cel
+/// [1, 2, 3].exists(x, x > 0) == true
+/// [{1:true, 2:true, 3:false}].exists(x, x > 0) == true
+/// ```
```

Comment:
This example was probably copied from above but not updated. Let's update for `exists_one`.

Reply:
Oh I forgot about that, have fixed that in the new commit.